HPCC-16438 Fix minor resource leak in configmgr reported by coverity

rpastrana opened the pull request:
- Deletes instDetails if new'ed prior to assigning it

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

rpastrana commented:
@ghalliday please review

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: a04be4b9c19024d006ff2bc4a308cbb986a42e58
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.934 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman commented:
Commit title should be something like "Fix minor resource leak in configmgr reported by coverity". The description of what code was actually changed has no place in the commit message (we can see that from the diff) - but the person reading the changelog has no idea from your title whether this is a bug that caused cores, leaks, wrong answers, exploding datacenters...

rpastrana committed:
HPCC-16438 Fix minor resource leak in configmgr reported by coverity

HPCC-16438 Fix minor resource leak in configmgr reported by coverity

- Delete possibly previously NEW'ed var before reassignment to avoid override
- Per code review comments, no need to check if NULL before delete
- Per code review comments, use delete directive rather than function
- Add comment explaining the unusual delete for ease of maintenance

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 781914513e48310cd3dcaa74d3954448f712d74d
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.573 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



HPCC Stop: OK
HPCC Uninstall: OK

ghalliday commented on deployment/deployutils/wizardInputs.cpp:
```diff
@@ -316,6 +316,9 @@ CInstDetails* CWizardInputs::getServerIPMap(const char* compName, const char* bu
       if(m_compIpMap.find(buildSetName) != NULL)
       {
+        if(instDetails != NULL)
+          delete(instDetails);
```
@rpastrana 
This is being picky, and would have ignored if only 1 of them, but a couple of issues
i) no need to check for null before calling delete.
ii) this looks like a function call rather than a use of the delete operator.

`delete instDetails;`

is much more common.

rpastrana commented:
@ghalliday please review.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 548e92bf1d9e006124f70b212ae43052487e7704
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.237 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



HPCC Stop: OK
HPCC Uninstall: OK

ghalliday commented:
@rpastrana please squash the commits.  Also a typo in your comment.

rpastrana commented:
@ghalliday I'll take a look for the typo, but why would we sqash the commits? Don't we lose code review change history? The reviewer can always view the net change from the "Files Changed" tab, and the merge into the target branch will result in a single commit anyway. Just curious about the squashing approach in this scenario. Thanks.

ghalliday commented:
@rpastrana there isn't generally any benefit in having the code review history in the source control system if it includes problems which never made it into the merged sources.  It would only confuse anyone looking at the history.  It is much better to have what is understood to be the correct change, than various stages along the way.
It is particularly true in this case where none of the original commit is actually kept in the subsequent commits.  You would have 3 commits, none of which kept anything from the previous change.

We do occasionally keep a review comments commit separate - to avoid going back around the cycle of squashing.  @richardkchapman should we still do this, or should we always squash?

The pull requests should retain the history of what has changed.

richardkchapman commented:
Generally I prefer the history / changelog to be clean rather than full of confusing commits that merely preserve mistakes/dead-ends that were never released.

It's useful to have multiple commits where a change was made in several independent steps, but not otherwise.

ghalliday commented:
@rpastrana

rpastrana commented:
@ghalliday

ghalliday commented:
@rpastrana it still has two commits and a typo (delet) in the comment

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 66e060eb2709c1f9f1fa1a008fcf32472a84c11f
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.374 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



HPCC Stop: OK
HPCC Uninstall: OK

jakesmith commented:
> We do occasionally keep a review comments commit separate - to avoid going back around the cycle of squashing.

Agree.
It generally greatly aides the reviewing process, if changes are not squashed between commit review requests, unless the change is so substantial that there is gain in seeing what's changed since the last review vs the overall change.


> Generally I prefer the history / changelog to be clean rather than full of confusing commits that merely preserve mistakes/dead-ends that were never released.

Agree.
Once the review is complete, the PR's history of changes that are mostly as a direct result of the review are noise and are better squashed before merging.

rpastrana commented:
@ghalliday

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: dca1df6412b1dec38dc3ccf550370f0044901a32
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.387 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: f9ea54f9e1f369d5a38a03d81c6ab5218bd905ef
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 93.571 seconds |


Install hpccsystems-platform-community_6.3.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |93 |93 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



HPCC Stop: OK
HPCC Uninstall: OK
